File: ta-java/RevenueProviderContract.java
import org.junit.Before;
import org.junit.Test;
import org.junit.Assert;

public abstract class RevenueProviderContract {


  private RevenueProvider revenueProvider;

  protected abstract RevenueProvider getRevenueProvider();

  @Before
  public void setUp() {
    revenueProvider = getRevenueProvider();
  }

  @Test
  public void revenue_provider_returns_revenue_for_valid_customer_id() {
  }

  @Test
  public void revenue_provider_throws_exception_for_invalid_customer_id() {
  }
}

File: ta-csharp/RevenueProviderContract.cs
using System;
using NUnit.Framework;


[TestFixture]
public abstract class RevenueProviderContract
{
    // private IRevenueProvider revenueProvider;

    protected abstract IRevenueProvider GetRevenueProvider();

    [SetUp]
    public void setup()
    {
        // revenueProvider = GetRevenueProvider();
    }

    [Test]
    public void revenue_provider_returns_revenue_for_valid_customer_id()
    {
    }

    [Test]
    public void revenue_provider_throws_exception_for_invalid_customer_id()
    {
    }
}
